apps/tools/inventory/fdsnxml2inv/convert2fdsnxml.cpp

```diff
@@ -368,12 +369,37 @@ void populateComments(const T1 *sc, T2 sx) {
 		}
 
 		FDSNXML::CommentPtr sx_comment = new FDSNXML::Comment;
-		int id;
-		if ( Core::fromString(id, comment->id()) )
+
+		if ( comment->id().substr(0, 16) == "FDSNXML:Comment/" ) {
+			IO::JSONArchive ar;
+
+			if (!ar.from(comment->text().c_str())) {
+				SEISCOMP_ERROR("failed to parse FDSNXML:Comment '%s'", comment->text().c_str());
+				continue;
+			}
+
+			int id = c+1;
+			string subject, value;
+			ar & NAMED_OBJECT_HINT("id", id, Core::Archive::STATIC_TYPE);
```

Comment:
Here `NAMED_OBJECT("id", id)` should be enough. Have you used `NAMED_OBJECT_HINT` because the simple version did not work?

Reply:
I don't know what the exact difference is.

NAMED_OBJECT() works with simple attributes like "id", but I get "BaseObject" instead of "contact" and "identifier":

<text>{"agency":"AWI Deutscher Geräte-Pool für amphibische Seismologie","webSite":"www.awi.de/depas","**BaseObject**":[{"name":["xxx"],"agency":["AWI-DEPAS"],"email":["xxx@xxx.org"]}]}</text>

<text>{"type":"Broadband seismometer","description":"Guralp CMG-40T 60s seismometer, DEPAS pool [config: SN-T4K39_2]","manufacturer":"Guralp Systems Ltd.","vendor":"Guralp Systems Ltd.","model":"Guralp CMG-40T 60s","serialNumber":"T4K39","**BaseObject**":[{"type":"hdl","value":"10881/sensor.a7561d1a-d518-475d-9733-30370432996c"}]}</text>

Reply:
`STATIC_TYPE` means that there is no polymorphism and the content of the serialized object is stored under the given name. Otherwise the classname must be stored somewhere and for that the attribute name is being used. For complex types it is OK to use `NAMED_OBJECT_HINT` but for simple types I would recommend the simpler version because of readability.